src/main/java/com/zaxxer/hikari/pool/PoolElf.java

```diff
@@ -281,15 +284,6 @@ void resetConnectionState(final PoolBagEntry poolEntry) throws SQLException
       }
    }
 
-   void resetPoolEntry(final PoolBagEntry poolEntry)
-   {
-      poolEntry.setReadOnly(isReadOnly);
-      poolEntry.setCatalog(catalog);
-      poolEntry.setAutoCommit(isAutoCommit);
-      poolEntry.setNetworkTimeout(networkTimeout);
-      poolEntry.setTransactionIsolation(transactionIsolation);
-   }
-
```

Comment:
I will grant that ``resetPoolEntry()`` is not clear.  When naming the method, it was originally was called from more than one place.

I would suggest reverting all of the changes above, and renaming this method to something like ``initializeDefaultState()`` or ``configureStateFromDefaults()``, ...

The ``PoolElf`` holds all of the pool defaults derived from the ``HikariConfig`` and after determining supported APIs, etc.  That is why it is responsible for ``resetConnectionState(PoolBagEntry)`` as well as ``resetPoolEntry(PoolBagEntry)``.  While the former is clearly named, the later is clearly not.